csi/deployment_util.go

```diff
@@ -180,19 +174,21 @@ func deploy(kubeClient *clientset.Clientset, obj runtime.Object, resource string
 		existingAnnos := existingMeta.GetAnnotations()
 		if annos[AnnotationCSIGitCommit] == existingAnnos[AnnotationCSIGitCommit] &&
 			annos[AnnotationCSIVersion] == existingAnnos[AnnotationCSIVersion] &&
-			annos[AnnotationKubernetesVersion] == existingAnnos[AnnotationKubernetesVersion] &&
 			existingMeta.GetDeletionTimestamp() == nil &&
 			!needToUpdateImage(existing, obj) {
 			// deployment of correct version already deployed
-			logrus.Infof("Detected %v %v CSI Git commit %v version %v Kubernetes version %v has already been deployed",
-				resource, name, annos[AnnotationCSIGitCommit], annos[AnnotationCSIVersion], annos[AnnotationKubernetesVersion])
+			logrus.Infof("Detected %v %v CSI Git commit %v version %v has already been deployed",
+				resource, name, annos[AnnotationCSIGitCommit], annos[AnnotationCSIVersion])
 			return nil
 		}
-	}
-	// otherwise clean up the old deployment
-	if err := cleanup(kubeClient, obj, resource, deleteFunc, getFunc); err != nil {
+		// otherwise clean up the old deployment
+		if err := cleanup(kubeClient, obj, resource, deleteFunc, getFunc); err != nil {
+			return err
+		}
+	} else if !apierrors.IsNotFound(err) {
```

Comment:
This is a nice little improvement. Thanks for adding it while in this section of code!